network/network.go

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (C) 2020. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package network
+
+import (
+	stdCrypto "crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/network/dag"
+	"github.com/nuts-foundation/nuts-node/network/log"
+	"github.com/nuts-foundation/nuts-node/network/p2p"
+	"github.com/nuts-foundation/nuts-node/network/proto"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"strings"
+	"sync"
+	"time"
+)
+
+// ModuleName defines the name of this module
+const ModuleName = "NetworkEngine"
+
+type NetworkEngine struct {
```

Comment:
Doc

Reply:
NetworkEngine is an network.Engine or just network.Network?